### ISSUE TITLE
Possible to select m4a instead of web with 'da'?

### ISSUE THREAD

**Issue by vnckppl** [mps-youtube/mps-youtube#389]
When downloading files from a playlist using 'da' often webm files are downloaded because they seem to be the best quality.

Is there an option to ignore the webm format and go straight to the next highest quality files (e.g. m4a)?

**Comment by ids1024** [comment]
Well, if you use `d` instead of `da`, you can manually select what format you want.

**Comment by vnckppl** [comment]
Yes, I know of this option. However, I am specifically looking to use da because I do not want to manually download x number of files and manually have to go over the options for each.

**Comment by ids1024** [comment]
Well then, it isn't possible yet, but could perhaps be added.

**Comment by pacoVela** [comment]
+1

**Comment by niwatoribaka** [comment]
If you are encoding all of your downloads -- e.g. with the `MP3 256K` preset -- you could find ` ~/.config/mps-youtube/transcode` file and add `webm` to the `valid for:` line. So that entry would look like...

```
name: MP3 256k
extension: mp3
valid for: ogg,m4a,webm
command: ENCODER_PATH -i IN -codec:a libmp3lame -b:a 256k OUT.EXT
```

I'm on a Mac using ffmpeg and this worked for me.  I forked this repo just today (to add a clever way for naming files and inserting metadata to downloaded albums so that I could drag a downloaded album right to iTunes) so I'm by no means an expert on mpsyt.

**Comment by vnckppl** [comment]
Encoding indeed recodes the audio stream. I am more interested to obtaining the most original highest quality non-webs files, although I understand that the difference in quality between that and a web-to-mp3 encoded file might be minimal.

**Comment by onoderat** [comment]
Hello niwatoribaka, 

Your fork sounds super cool! Do you plan to merge it at some point? If not, could you make the forked repo public, so I can download it?

Thanks!

**Comment by gitsper** [comment]
@onoderat So since playing with mps-youtube a year ago, I've moved all of my repos&forks to the account gitsper.  You can see my fork [here](https://github.com/gitsper/mps-youtube) (should be public).  I think at this point I'm a bit too far behind to merge, but I may post a feature request and see if there is interest in having this functionality on the main branch.

**Comment by Atcold** [comment]
Alright, set a PR with this edit https://github.com/mps-youtube/mps-youtube/pull/1065
